jsonrpc/server.go

```diff
@@ -300,6 +304,19 @@ func buildArguments(params, handler any, configuredParams []Parameter) ([]reflec
 			return reflect.ValueOf(nil), err
 		}
 
+		// Get the underlying value, dereferencing until we reach the "raw" kind.
+		value := handlerParam.Elem()
+		for value.Kind() == reflect.Pointer {
+			value = value.Elem()
```

Comment:
this is not safe, we might get a nil pointer after we deference it couple of times. I dont think we need to deference it multiple times, we can just assume there wont be pointers to pointers in any of the arguments.

Validator deferences the value if it is a pointer anyways.
https://github.com/go-playground/validator/blob/c8b98cbcea08971fbead222ddf5fd406a40d40ed/validator_instance.go#L373-L375

to keep things simple here we can call validate.Struct without doing any checks whether if the argument is a struct or not. If it is not a struct validator will return InvalidValidationError and we can just silently ignore that

https://github.com/go-playground/validator/blob/v10.13.0/validator_instance.go#LL378C11-L378C33